linux/x86-64/python3/acrcloud/async_recognizer.py

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env python
+#-*- coding:utf-8 -*-
+
+'''
+    @author qinxue.pan E-mail: xue@acrcloud.com
+    @version 1.0.0
+    @create 2015.10.01
+'''
+
+import os
+import sys
+import hmac
+import time
+import json
+import base64
+import hashlib
+import aiohttp
+import async_timeout
+import datetime
+
+import acrcloud_extr_tool
+
+'''
+Copyright 2015 ACRCloud Recognizer v1.0.0
+
+This module can recognize ACRCloud by most of audio/video file. 
+        Audio: mp3, wav, m4a, flac, aac, amr, ape, ogg ...
+        Video: mp4, mkv, wmv, flv, ts, avi ...
+
+Example:
+    config = {
+        'host':'ap-southeast-1.api.acrcloud.com',
+        'access_key':'XXXXXXXX',
+        'access_secret':'XXXXXXXX',
+        'timeout':5
+    }
+    re = ACRCloudRecognizer(config)
+
+    #recognize by file path, and skip 180 seconds from from the beginning of "aa.mp3".
+    print re.recognize_by_file('aa.mp3', 180)
+
+    buf = open('aa.mp3', 'rb').read()
+    #recognize by file_audio_buffer that read from file path, and skip 180 seconds from from the beginning of "aa.mp3".
+    print re.recognize_by_filebuffer(buf, 180)
+
+    #aa.wav is (RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, mono 8000 Hz)
+    buf = open('aa.wav', 'rb').read()
+    buft = buf[1024000:192000+1024001]
+    recognize by audio_buffer(RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, mono 8000 Hz)
+    print re.recognize(buft)
+'''
+class ACRCloudRecognizeType:
+    ACR_OPT_REC_AUDIO = 0  # audio fingerprint
+    ACR_OPT_REC_HUMMING = 1 # humming fingerprint
+    ACR_OPT_REC_BOTH = 2 # audio and humming fingerprint
+
+class ACRCloudRecognizer:
+    def __init__(self, config, loop):
+        self.loop = loop
+        self.config = config
+        self.host = config.get('host', 'ap-southeast-1.api.acrcloud.com')
+        self.query_type = config.get('query_type', 'fingerprint')
+        self.access_key = config.get('access_key')
+        self.access_secret = config.get('access_secret')
+        self.timeout = config.get('timeout', 5)
+        self.recognize_type = config.get('recognize_type', ACRCloudRecognizeType.ACR_OPT_REC_AUDIO)
+        if self.recognize_type > 2 or self.recognize_type < 0:
+            self.recognize_type = ACRCloudRecognizeType.ACR_OPT_REC_AUDIO
+        self.debug = config.get('debug', False)
+        if not self.access_key or not self.access_secret:
+            print('recognize init(none access_key or access_secret)')
+            sys.exit(1)
+
+        if self.debug:
+            acrcloud_extr_tool.set_debug()
+
+    async def post_multipart(self, url, fields, files, timeout):
+        content_type, body = self.encode_multipart_formdata(fields, files)
+
+        if not content_type and not body:
+            return ACRCloudStatusCode.get_result_error(ACRCloudStatusCode.HTTP_ERROR_CODE,
+                                                       'encode_multipart_formdata error')
```

Comment:
I would leave this line alone so that it does not show up as a diff from the original.